lib/matplotlib/tests/test_pickle.py

```diff
@@ -88,16 +89,21 @@ def _generate_complete_test_figure(fig_ref):
 
     plt.subplot(3, 3, 9)
     plt.errorbar(x, x * -0.5, xerr=0.2, yerr=0.4)
+    plt.legend(draggable=True)
 
 
 @mpl.style.context("default")
 @check_figures_equal(extensions=["png"])
 def test_complete(fig_test, fig_ref):
     _generate_complete_test_figure(fig_ref)
     # plotting is done, now test its pickle-ability
-    pkl = BytesIO()
-    pickle.dump(fig_ref, pkl, pickle.HIGHEST_PROTOCOL)
-    loaded = pickle.loads(pkl.getbuffer())
+    pkl = pickle.dumps(fig_ref, pickle.HIGHEST_PROTOCOL)
+    # FigureCanvasAgg is picklable, but GUI canvas are generally not, so we make sure
+    # that it doesn't end up it the pickle.  In order to keep the test independent of
+    # GUI toolkits, just check that there's no reference to FigureCanvasAgg in the
+    # pickle stream.
+    assert "FigureCanvasAgg" not in [arg for op, arg, pos in pickletools.genops(pkl)]
```

Comment:
I'm a bit confused; isn't the default in tests to use Agg?

Reply:
Yes.  The point is that 1) I don't want the test to need spinning up a GUI toolkit, so we're stuck with running it with Agg, and 2) I want to check that no canvas ends up being pickled (even though FigureCanvasAgg is indeed picklable), so I instead check for the absence of FigureCanvasAgg in the pickle stream (prior to this PR the check would have failed).

Reply:
OK, that didn't seem clear from the comment. I thought the 'it' in "it doesn't end up it (sic) the pickle" was the GUI canvas, which is always true since the default is Agg.

So I suggest something like:
```suggestion
    # FigureCanvasAgg is picklable and GUI canvas are generally not, but there should
    # be no reference to the canvas in either case. In order to keep the test independent of
    # GUI toolkits, run with Agg, and check that there's no reference to FigureCanvasAgg in the
    # pickle stream.
    assert "FigureCanvasAgg" not in [arg for op, arg, pos in pickletools.genops(pkl)]
```

Reply:
sure, done.